bluemix/motd/motd.go

```diff
@@ -0,0 +1,105 @@
+package motd
+
+import (
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix"
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/api"
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/configuration/core_config"
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/bluemix/terminal"
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/common/rest"
+	"github.com/IBM-Cloud/ibm-cloud-cli-sdk/plugin"
+)
+
+type MODMessages struct {
+	VersionRange string `json:"versionRange"`
+	Region       string `json:"region"`
+	OS           string `json:"os"`
+	Message      string `json:"message"`
+}
+
+type MODResponse struct {
+	Messages []MODMessages `json:"messages"`
+}
+
+func CheckMessageOftheDayForPlugin(pluginConfig plugin.PluginConfig) bool {
+	currentMessagOfTheDay := pluginConfig.Get("MessageOfTheDay")
+	if currentMessagOfTheDay == nil {
+		return false
+	}
+	if currentMessagOfTheDayTimestamp, ok := currentMessagOfTheDay.(string); ok {
+		lastCheckTime, parseErr := strconv.ParseInt(currentMessagOfTheDayTimestamp, 10, 64)
+		if parseErr != nil {
+			return false
+		}
+		return time.Since(time.Unix(lastCheckTime, 0)).Hours() < 24
+	}
+	return false
+}
+
+func CheckMessageOfTheDay(client *rest.Client, config core_config.ReadWriter, pluginConfig plugin.PluginConfig, modURL string, ui terminal.UI, version string) {
+	// the pluginConfig variable will be cast-able to a pointer to a plugin config type if the display is for a plugin
+
+	if config != nil {
+		if !config.CheckMessageOfTheDay() {
+			return
+		}
+		defer config.SetMessageOfTheDayTime()
+	} else {
+		if !CheckMessageOftheDayForPlugin(pluginConfig) {
+			return
+		}
+		defer pluginConfig.Set("MessageOfTheDayTime", time.Now().Unix())
```

Comment:
There needs to a nil check here since this can be null.

Reply:
Are you talking about the `pluginConfig` variable? If `config` is nil, then `pluginConfig`--in all valid uses of this function--needs to be non-nil. 
I suppose it is possible for there to be a misuse of this function, but then we would probably also have to add some defensive programming on line 49. Personally, I don't think it's necessary, as we will have very specific prescriptions for how this function is used (we might even exactly control it, pending discussion with SDK/plugin teams) and we will write the source code for the caller on the core CLI.

Reply:
Hmm I am not sure why I said that we should do a null check here. We do need to check for an error before we defer
```suggestion
		defer func() error {
		   if err := pluginConfig.Set("MessageOfTheDayTime", time.Now().Unix()); err != nil {
		       return err
		   }
		   return nil
	}()
```

Reply:
Gotcha. Yes, error-checking is wise, but nullcheck is unnecessary as described above. I'm committing your suggestion.